src/targets/gpu/mlir.cpp

```diff
@@ -677,12 +674,9 @@ struct mlir_program
 
     void set_tuning(const value& v)
     {
-        auto str = v.to<std::string>();
-        // We need to make a copy of the buffer since mlirRockTuningSetFromStr may modify the string
-        std::vector<char> buffer(str.begin(), str.end());
-        buffer.push_back(0);
-        if(not mlirRockTuningSetFromStr(mmodule.get(), buffer.data()))
-            MIGRAPHX_THROW("Failed setting tuning key: " + str);
+        auto str = v.to<std::string_view>();
```

Comment:
`string_view` wont work if `v` is not a string, and it may crash in a non-obvious way. Instead use `if_string` method and throw when its null:

```cpp
auto* str = v.if_string();
if(not str)
    MIGRAPHX_THROW("Only string is valid.");
if(not mlirRockTuningSetFromStr(mmodule.get(), make_mlir_string_ref(*str)))
    ...
```